### ISSUE TITLE
tar command fails inside docker for Java buildpack

### ISSUE THREAD

**Issue by jillson** [deis/deis#4930]
I'm seeing similar reports for Meteor buildpack to docker and coreos (https://github.com/docker/docker/issues/19647, https://github.com/coreos/bugs/issues/1095), and so I'm not sure if i should be reporting this there or here, but at any rate:

As of the last couple of weeks, when I attempt to push a Java-based application (which had successfully deployed on deis previousily), I am now getting

tar: jre/lib/amd64/server: Directory renamed before its status could be extracted
tar: jre/lib/amd64: Directory renamed before its status could be extracted
tar: jre/lib: Directory renamed before its status could be extracted
tar: jre: Directory renamed before its status could be extracted
tar: Exiting with failure status due to previous errors

Doing a little poking around, It's failing on the command `tar pxzf /tmp/build/.jdk/jdk.tar.gz -C /tmp/build/.jdk` which gets called by the Heroku java buildpack

Bottom line: should I provide more details on deis/docker/core/etc. versions or should I be lookup upstream (i.e. to docker or core os)?  The bug reports I linked to above suggest it was related to some overlayfs but I don't know why that would suddenly change since I haven't knowingly updated deis packages etc. at the point when stuff started breaking.

**Comment by jgmize** [comment]
This seems like it may be related to the same overlayfs bug discussed in https://github.com/docker/docker/issues/12327 and https://bugzilla.kernel.org/show_bug.cgi?id=109611.

**Comment by bacongobbler** [comment]
thank you for the sleuthing, @jgmize. It does indeed look like we're gonna have to wait for an upstream fix from Docker. @jillson I'd suggest trying to build your project via Dockerfile for now using the workarounds explained in docker/docker#12327 until a fix comes up and is blessed by CoreOS.

**Comment by jillson** [comment]
@backgobbler.  Thanks.  I'm a bit concerned the way I'm using deis as a backend would seem to require some workarounds that I'm not liking, but looking at the upstream links, it sounds like we don't have a good ETA for when the "real" fix will get shipped etc.

 Just to check: any reason I shouldn't be able (on a "test" version of deis) rebuild the overlayfs module with that patch (I'm guessing I probably will need to do the coreOS upgrade described on the upgrading deis page to get "close enough" to the kernel version that patch targets and do some fun stuff to get the right patched source etc. to build against on yet another system).  Doing lsmod on my current deis nodes, I see overlay, which hopefully means I don't have to actually rebuild an entire patched kernel but rather just replace the overlay.ko (and presumably the initrd stuff that has overlay.ko baked in for bootstrapping reasons)

(While I have fond memories of getting to patch Linux 2.0.X and eagerly recompiling each new released build to see if the minor tweak needed to support my offbrand tulip-clone network card finally "stuck" in the mainline, I'd just as soon avoid trying to replace the kernel / boot / etc.)

**Comment by jillson** [comment]
(So I figured out how to build a new overlay.ko file in a docker container, but then ran into the issue that I'm not totally sure how to update the module and reboot the kernel to pick it up without the AWS stuff deciding to terminate my VM and start a new one or otherwise change things that breaks Deis)... and then there's the issue that something else went wrong with Deis such that the build process wouldn't start... and so my net solution was to build another deis cluster and migrate stuff to that... and so far (*knock on wood*) this new cluster hasn't has the overlay issue.  Going to mark this closed since I can longer reproduce anything.
